### ISSUE TITLE
for snapper could hook in plugins

### ISSUE THREAD

**Issue by bastien-roucaries** [Antynea/grub-btrfs#314]
Hi,

I really believe for snapper a good configuration is plugins one
create-snapshot-post subvolume fstype snapshot-number
delete-snapshot-post subvolume fstype snapshot-number

Could you please give me some guidance of how achieve this kind of modification ?

Thanks

Bastien

**Comment by bastien-roucaries** [comment]
@Schievel1 can you get a glimse ?

**Comment by bastien-roucaries** [comment]
See https://man.archlinux.org/man/snapper.8.en#PLUGINS

**Comment by Schievel1** [comment]
ok, I read that but I am still a bit confused what you want to do. 
Do you want to create a new grub submenu whenever snapper makes a new snapshot? This is actually what `grub-btrfsd` is for, but you could do something like 
```
create_grub_menu() {
    #  create the grub submenu of the whole grub menu, depending on whether the submenu already exists
    #  and gives feedback if it worked
    if grep "snapshots-btrfs" "${GRUB_BTRFS_GRUB_DIRNAME:-/boot/grub}"/grub.cfg; then
    	  if  /etc/grub.d/41_snapshots-btrfs; then
		        echo "Grub submenu recreated"
          else
		      echo "[!] Error during grub submenu creation (grub-btrfs error)"
          fi
    else
        if ${GRUB_BTRFS_MKCONFIG:-grub-mkconfig} -o "${GRUB_BTRFS_GRUB_DIRNAME:-/boot/grub}"/grub.cfg; then
		    echo "Grub menu recreated"
	    else
		    echo "[!] Error during grub menu creation (grub/ grub-btrfs error)"
	    fi
    fi
}

if [[ $1 == create-snapshot-post ]]; then
    create_grub_menu
fi
```

Something like that?

**Comment by aidan-gibson** [comment]
I think they're essentially referring to what these accomplish:
https://github.com/wesbarnett/snap-pac
https://github.com/maximbaz/snap-pac-grub

But I literally just use grub-btrfsd, much better solution imo

**Comment by bastien-roucaries** [comment]
@Schievel1 yes something like this snapper could be send in modern language **push notification** every time it create an interesting notification.

You use here a full hammer that is every time you get a notification, you recreate the whole menu, but if you could run grub-btrfs special hook it could be more efficient
